### PR TITLE
adds form submission handler to contact form

### PR DESCRIPTION
**Pull request by sophielevens**
+ Adds .js form submission to contact form 
+ Adds if statement to BETA form handler 

[References this issue](https://github.com/dwyl/contact/issues/6)

### CODE REVIEW

File: js/app_contact_form_handler.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// get all data in form and return object
+function getContactFormData() {
```
Comment (nelsonic) [inline js/app_contact_form_handler.js]: we should not be copy-pasting the form handler code when we can "include" it directly from the CDN using https://rawgit.com to keep things "DRY".

e.g: https://cdn.rawgit.com/dwyl/learn-to-send-email-via-google-script-html-no-server/faa599f47774ca6397ece7d57a08a247f0469186/form-submission-handler.js
For a _specific_ version of the code.

Or without the Git Hash (_if we always want to use the latest version of the code_):
https://cdn.rawgit.com/dwyl/learn-to-send-email-via-google-script-html-no-server/master/form-submission-handler.js

**Comment by nelsonic** [comment]
@rub1e let me know if you have time to "remote pair" on getting this finished sometime this week. thanks! 😉

**Comment by rub1e** [comment]
Tell you what mate @nelsonic, seeing as you're on holiday and I'm not doing any dev work on Killer Football this week (and because this is an open-source repo 😉) I'll do it `pro bono` in the next couple of days (thursday morning most likely)

Bear with the spam for another 48 hours!

Or the alternative would be to merge this now and I can replace with the CDN version on thursday?

**Comment by rub1e** [comment]
OK @nelsonic, if you will be so kind as to indulge my ignorance of basic < *squints* > coding...

Here is what I think is wrong:
* The honeypot was never going to work because
  * There wasn't a honeypot `input` in the first place
  * The `validateHuman` function hadn't been copied into the form submission handler JS file (nor had the `validEmail` function)
  * And actually the local JS files weren't handling the form submissions anyway - as the script was being loaded via `rawgit` anyway (see [this line](https://github.com/dwyl/dwyl-site/blob/master/index.html#L212))
  * Including the script in this way won't work, because the honeypot function is an optional extra that you have to `uncomment` (as an aside, I don't get the point of this - surely you can control the honeypot function just by not including a honeypot `input`??)
  * Even if the honeypot were included, we still couldn't use that script, because it's hard-coded for a `gform` element and we've got two different forms (contact and beta signup)

* So why (I wondered) is the beta signup form actually working?
  * Well, that script is actually loaded from our internal JS file, so why *wouldn't* it work?
  * Well, it wouldn't work because that JS file is also missing `validEmail` and `validateHuman` functions, and so it should throw a `referenceError`
  * BUT, it actually does work because both of those functions are included from external script 
  * EXCEPT, the email validation doesn't work because it reveals a hidden error message which isn't actually included in our form

So what I've done is just make it work - it's not DRY at this stage because I can't see an obvious way of making it DRY without rewriting a fair portion of the script - but I'll make a few comments as a `review` to see if I've got the right end of the stick

Grateful for your comments as it took me quite a while to twig what was going on (I've become very used to the ES6 module import/export system)

File: time-app.html [hunk 2]
```diff
@@ -98,6 +98,9 @@ <h3 class="white f3">Become a BETA tester!</h3>
               <input id="app_email" name="app_email" class="br2 ba dwyl-b--dark-gray pa2 w-100 border-box" type="email" value="" required placeholder="tim@cern.com" />
```
Comment (rub1e) [inline time-app.html]: Does this actually need an `id`? 

Could it just have `name="email"`, making it easier to DRY up the handler?

Comment (nelsonic) [reply time-app.html]: I think it's only defined that way because there was a page where there were two forms ...
can be changed in next iteration. 👍

**Comment by rub1e** [comment]
I did an acceptable code!

**Pull request merged**